src/Core/MailTemplate/MailTemplateGenerator.php

```diff
@@ -0,0 +1,152 @@
+<?php
+/**
+ * 2007-2019 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\MailTemplate;
+
+use PrestaShop\PrestaShop\Core\Exception\FileNotFoundException;
+use PrestaShop\PrestaShop\Core\Exception\InvalidArgumentException;
+use PrestaShop\PrestaShop\Core\Language\LanguageInterface;
+use Psr\Log\LoggerAwareTrait;
+use Psr\Log\NullLogger;
+use Symfony\Component\Filesystem\Filesystem;
+
+/**
+ * Class MailTemplateGenerator iterates through the layouts in the provided catalog,
+ * it uses the Renderer to display them (with the requested LanguageInterface) and
+ * then export them as template files in the specified output folder.
+ */
+class MailTemplateGenerator
+{
+    use LoggerAwareTrait;
```

Comment:
Because traits are bad (:nauseated_face:) and because the LoggerAwareTrait only have one method, you can create and interface or an abstact class instead which implement the setLogger method?
Or add the $logger in constructor.

Reply:
I specifically didn't want to have the logger in my constructor. Because the class doesn't need it at all to work. It was only used as a debug tool, so I don't want to make the constructor nor the test more complicated.

That's why I preferred the setter injection in this case. As to the `LoggerAwareTrait` it is available in PSR code, why the hell would I bother implementing it myself? You guys really have issues with traits ^^

Reply:
No! I don't want to complicate this PR even more just because of a trait issue.
And I don't see the interest of creating a dev class, this can only bring unexpected problem (the dev class starting to behave differently from the prod one, or what else?)
Either we keep the trait or I implement it manually but I won't clone this class.